riscv/build.rs

```diff
@@ -1,12 +1,13 @@
 use std::env;
 
 fn main() {
-    let target = env::var("TARGET").unwrap();
+    let target = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+    let ptr_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap();
 
-    if target.starts_with("riscv32") {
+    if target.contains("riscv") && ptr_width == "32" {
         println!("cargo:rustc-cfg=riscv");
         println!("cargo:rustc-cfg=riscv32");
-    } else if target.starts_with("riscv64") {
+    } else if target.contains("riscv") && ptr_width == "64" {
```

Comment:
It would be better to compare a full `target_arch` instead of referring to a `target_pointer_width`. (It is more simple and there is also a [proposal on 32-bit (pointer) ABI of riscv64](https://lwn.net/Articles/951187/).)

```rs
    let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();

    if target_arch == "riscv32" {
        println!("cargo:rustc-cfg=riscv");
        println!("cargo:rustc-cfg=riscv32");
    } else if target_arch == "riscv64" {
```